assets/js/modules/idea-hub/datastore/idea-state.js

```diff
@@ -32,23 +32,63 @@ const { receiveError, clearError } = errorStoreActions;
 
 const fetchPostUpdateIdeaStateStore = createFetchStore( {
 	baseName: 'updateIdeaState',
-	controlCallback: ( { name, saved, dismissed } ) => {
+	controlCallback: ( { name, saved, dismissed } ) => new Promise( async ( resolve, reject ) => {
 		const params = { name };
 
 		if ( saved !== undefined ) {
 			params.saved = saved;
 		} else {
 			params.dismissed = dismissed;
 		}
-		return API.set( 'modules', 'idea-hub', 'update-idea-state', params );
-	},
+
+		try {
+			const results = await API.set( 'modules', 'idea-hub', 'update-idea-state', params );
+			await API.invalidateCache( 'modules', 'idea-hub', 'new-ideas' );
+			await API.invalidateCache( 'modules', 'idea-hub', 'saved-ideas' );
+
+			resolve( results );
+		} catch ( err ) {
+			reject( err );
+		}
```

Comment:
Actually, I think we can simplify it once again and get rid of `new Promise` approach. How about doing this?

```javascript
controlCallback: ( { name, saved, dismissed } ) => {
    const params = { name };

    if ( saved !== undefined ) {
        params.saved = saved;
    } else {
        params.dismissed = dismissed;
    }

    return API.set( 'modules', 'idea-hub', 'update-idea-state', params ).then( () => async {
        await API.invalidateCache( 'modules', 'idea-hub', 'new-ideas' ).catch( () => {} );
        await API.invalidateCache( 'modules', 'idea-hub', 'saved-ideas' ).catch( () => {} );
    } );
},
```

This should also work, shouldn't it? Sorry, for the incorrect suggestion at the beginning.

Reply:
Yep, works with a few tweaks from your suggestion :)

Reply:
Ah... yes, you are right, we need to return the result :laughing: